In [1]:
data = range(4)

In [2]:
data

range(0, 4)

In [3]:
list(data)

[0, 1, 2, 3]

In [11]:
%matplotlib inline
import pandas as pd
import seaborn as sns
import math
import matplotlib.pyplot as plt
import numpy as np
import os
import keyword
import csv
import sys
import scipy
import zipfile
import datetime
import random

In [10]:
current_directory = os.getcwd
print("Current Working Directory")

Current Working Directory


In [14]:
CME = pd.read_csv('data/CME.csv')
EQR = pd.read_csv('data/EQR.csv')
FISV = pd.read_csv('data/FISV.csv')
GE = pd.read_csv('data/GE.csv')
LNC = pd.read_csv('data/LNC.csv')



In [19]:
CME.head(5)

Date        High         Low        Open       Close     Volume  \
0 2017-11-13  140.350006  137.509995  137.720001  140.149994  1142100.0   
1 2017-11-14  141.039993  138.929993  139.679993  140.199997  1267500.0   
2 2017-11-15  141.440002  139.509995  139.720001  139.669998  1017300.0   
3 2017-11-16  140.610001  139.259995  140.380005  140.089996   740800.0   
4 2017-11-17  140.130005  139.039993  139.570007  139.490005   924200.0   

    Adj Close  
0  124.103569  
1  124.147797  
2  123.678535  
3  124.050438  
4  123.519112

In [28]:
firms = ['CME', 'EQR', 'FISV', 'GE', 'LNC']

In [30]:
CME['Date'] = pd.to_datetime(CME['Date'])
EQR['Date'] = pd.to_datetime(EQR['Date'])
FISV['Date'] = pd.to_datetime(FISV['Date'])
GE['Date'] = pd.to_datetime(GE['Date'])
LNC['Date'] = pd.to_datetime(LNC['Date'])



In [29]:
for f in firms:
    f['Date'] = pd.to_datetime(f['Date'])


TypeError: string indices must be integers, not 'str'

In [31]:
CME.head()

Date        High         Low        Open       Close     Volume  \
0 2017-11-13  140.350006  137.509995  137.720001  140.149994  1142100.0   
1 2017-11-14  141.039993  138.929993  139.679993  140.199997  1267500.0   
2 2017-11-15  141.440002  139.509995  139.720001  139.669998  1017300.0   
3 2017-11-16  140.610001  139.259995  140.380005  140.089996   740800.0   
4 2017-11-17  140.130005  139.039993  139.570007  139.490005   924200.0   

    Adj Close  
0  124.103569  
1  124.147797  
2  123.678535  
3  124.050438  
4  123.519112

In [39]:
type(CME['2017-11-13'])

KeyError: '2017-11-13'

In [26]:
EQR.head()

Date       High        Low       Open      Close     Volume  Adj Close
0 2017-11-13  70.250000  68.800003  69.150002  70.150002  1377500.0  60.831173
1 2017-11-14  70.349998  69.570000  70.089996  69.980003  1520100.0  60.683773
2 2017-11-15  70.089996  68.900002  69.970001  69.029999  1072900.0  59.859966
3 2017-11-16  69.419998  68.529999  69.070000  69.220001  1025500.0  60.024731
4 2017-11-17  69.300003  68.169998  68.980003  68.320000  1630400.0  59.244286

In [32]:
firms

['CME', 'EQR', 'FISV', 'GE', 'LNC']

In [33]:
FISV.head()

Date       High        Low       Open      Close     Volume  Adj Close
0 2017-11-13  64.285004  63.689999  63.985001  64.080002  1387000.0  64.080002
1 2017-11-14  64.925003  63.759998  63.900002  64.864998  1428600.0  64.864998
2 2017-11-15  64.915001  64.000000  64.764999  64.260002  1989200.0  64.260002
3 2017-11-16  65.050003  64.360001  64.455002  64.769997  1337000.0  64.769997
4 2017-11-17  64.739998  63.959999  64.720001  64.195000  1611000.0  64.195000

In [34]:
GE.head()

Date        High         Low        Open       Close      Volume  \
0 2017-11-13  159.615387  144.230774  155.846161  146.307693  34002462.0   
1 2017-11-14  145.230774  134.307693  144.538467  137.692307  40632384.0   
2 2017-11-15  141.384613  134.615387  134.692307  140.461533  18767840.0   
3 2017-11-16  142.153839  138.846161  142.153839  140.384613  11618958.0   
4 2017-11-17  142.076920  140.076920  141.076920  140.076920   7859436.0   

    Adj Close  
0  145.114578  
1  136.569443  
2  139.316086  
3  139.239792  
4  138.934586

In [36]:
LNC.head()

Date       High        Low       Open      Close     Volume  Adj Close
0 2017-11-13  74.139999  73.010002  73.110001  73.900002  1060500.0  64.800636
1 2017-11-14  73.510002  72.760002  73.260002  73.440002  1150100.0  64.397285
2 2017-11-15  74.400002  72.540001  72.680000  73.949997  1484200.0  64.844482
3 2017-11-16  74.970001  73.570000  74.580002  74.010002  1418700.0  64.897095
4 2017-11-17  74.739998  73.470001  73.470001  74.279999  1100400.0  65.133835

1. After I checked the data to see each company's structure, I noticed immediately, that we can observe that only company FISV is closing on the same value.

The adjusted closing price contains accurate indication of the stock's value unlike the closing price (how much investors paid for shares at the end of a trading day) 

In [63]:
for i in firms:
    highest_adj_close = i.groupby('i')['Adj Close'].max()
    lowest_adj_close = i.groupby('i')['Adj Close'].min()

    print(highest_adj_close)
    print(lowest_adj_close)

AttributeError: 'str' object has no attribute 'groupby'

In [49]:
highest_adj_close = CME['Adj Close'].max()

lowest_adj_close = CME['Adj Close'].min()

print(highest_adj_close)
print(lowest_adj_close)

250.75999450683597
123.51911163330078


In [50]:
highest_adj_close = EQR['Adj Close'].max()

lowest_adj_close = EQR['Adj Close'].min()

print(highest_adj_close)
print(lowest_adj_close)

93.23999786376952
44.40938949584961


In [51]:
highest_adj_close = FISV['Adj Close'].max()

lowest_adj_close = FISV['Adj Close'].min()

print(highest_adj_close)
print(lowest_adj_close)

126.5500030517578
63.85499954223633


In [52]:
highest_adj_close = GE['Adj Close'].max()

lowest_adj_close = GE['Adj Close'].min()

print(highest_adj_close)
print(lowest_adj_close)

145.3667449951172
43.81699752807617


In [53]:
highest_adj_close = LNC['Adj Close'].max()

lowest_adj_close = LNC['Adj Close'].min()

print(highest_adj_close)
print(lowest_adj_close)

75.65805053710938
15.741825103759766


2. in those previous tables it can be seen the max and min of the adj close of each firms

In [83]:
CME['Log Return'] = np.log(CME['Adj Close'] / CME['Adj Close'].shift(1))

highest_gain_day = CME['Log Return'].idxmax()
highest_loss_day = CME['Log Return'].idxmin()

In [84]:
print(highest_gain_day)
print(highest_loss_day)
CME['Log Return'].describe()

594
586


count    1257.000000
mean        0.000270
std         0.018176
min        -0.203207
25%        -0.007453
50%         0.000903
75%         0.008226
max         0.111902
Name: Log Return, dtype: float64

In [85]:
firm = EQR
firm['Log Return'] = np.log(firm['Adj Close'] / firm['Adj Close'].shift(1))

highest_gain_day = firm['Log Return'].idxmax()
highest_loss_day = firm['Log Return'].idxmin()
print(highest_gain_day)
print(highest_loss_day)
firm['Log Return'].describe()

752
586


count    1258.000000
mean        0.000025
std         0.018368
min        -0.185287
25%        -0.008015
50%         0.000599
75%         0.008373
max         0.140302
Name: Log Return, dtype: float64

In [86]:
firm = FISV
firm['Log Return'] = np.log(firm['Adj Close'] / firm['Adj Close'].shift(1))

highest_gain_day = firm['Log Return'].idxmax()
highest_loss_day = firm['Log Return'].idxmin()
print(highest_gain_day)
print(highest_loss_day)
firm['Log Return'].describe()

592
586


count    1257.000000
mean        0.000358
std         0.019345
min        -0.182360
25%        -0.008412
50%         0.000805
75%         0.009416
max         0.118072
Name: Log Return, dtype: float64

In [87]:
firm = GE
firm['Log Return'] = np.log(firm['Adj Close'] / firm['Adj Close'].shift(1))

highest_gain_day = firm['Log Return'].idxmax()
highest_loss_day = firm['Log Return'].idxmin()
print(highest_gain_day)
print(highest_loss_day)
firm['Log Return'].describe()

592
586


count    1258.000000
mean       -0.000413
std         0.027478
min        -0.164394
25%        -0.013760
50%        -0.000965
75%         0.013300
max         0.137411
Name: Log Return, dtype: float64

In [81]:
firm = LNC
firm['Log Return'] = np.log(firm['Adj Close'] / firm['Adj Close'].shift(1))

highest_gain_day = firm['Log Return'].idxmax()
highest_loss_day = firm['Log Return'].idxmin()
print(highest_gain_day)
print(highest_loss_day)
firm['Log Return'].describe()

592
1252


count    1258.000000
mean       -0.000484
std         0.035288
min        -0.402686
25%        -0.013548
50%         0.000897
75%         0.014970
max         0.275359
Name: Log Return, dtype: float64

3. I calculated those logarithmic returns and made describe function

4. Four firms (all besides EQR) got their gain near 600, while EQR achieved 752. The biggest loss has LNC with 1252. 

In [103]:
ax = czech_cases.plot(color = 'lightgrey',label='other values',legend=True,**plotconfig)
czech_cases.loc[(czech_cases['new_cases'] >= 5000) & (czech_cases['new_cases'] < 15000), 'new_cases'].plot(ax=ax,label='Values between 500 and 750',legend=True,**plotconfig)
#czech_cases.loc[czech_cases.index.weekday == 6, 'new_cases'].plot(ax=ax,label='Sunday',legend=True,**plotconfig)
#czech_cases.loc[czech_cases.index.weekday == 5,'new_cases'].plot(ax=ax,label='Saturday',legend=True,**plotconfig)

NameError: name 'czech_cases' is not defined

this would be my source to make plots

In [57]:
sap = pd.read_csv('data/companies_no_subindustry.csv')

In [58]:
sap.head()

Symbol     Security  GICS_Sector                       hq    included  \
0    MMM           3M  Industrials    Saint Paul, Minnesota  09.08.1976   
1    AOS  A. O. Smith  Industrials     Milwaukee, Wisconsin  26.07.2017   
2    ABT       Abbott  Health Care  North Chicago, Illinois  31.03.1964   
3   ABBV       AbbVie  Health Care  North Chicago, Illinois  31.12.2012   
4   ABMD      Abiomed  Health Care   Danvers, Massachusetts  31.05.2018   

       Founded  
0         1902  
1         1916  
2         1888  
3  2013 (1888)  
4         1981

In [88]:
sap.isna()==1

Symbol  Security  GICS_Sector     hq  included  Founded
0     False     False        False  False     False    False
1     False     False        False  False     False    False
2     False     False        False  False     False    False
3     False     False        False  False     False    False
4     False     False        False  False     False    False
..      ...       ...          ...    ...       ...      ...
498   False     False        False  False     False    False
499   False     False        False  False     False    False
500   False     False        False  False     False    False
501   False     False        False  False     False    False
502   False     False        False  False     False    False

[503 rows x 6 columns]

In [89]:
(sap['included'].isna()).sum()#or cumprod() or maby not

45

2.1. we found 45 companies without filling

In [91]:
sap.dropna(subset=['included'], inplace=True)


In [92]:
(sap['included'].isna()).sum()#or cumprod() or maby not

0

In [95]:
sap.head()

Symbol     Security  GICS_Sector                       hq    included  \
0    MMM           3M  Industrials    Saint Paul, Minnesota  09.08.1976   
1    AOS  A. O. Smith  Industrials     Milwaukee, Wisconsin  26.07.2017   
2    ABT       Abbott  Health Care  North Chicago, Illinois  31.03.1964   
3   ABBV       AbbVie  Health Care  North Chicago, Illinois  31.12.2012   
4   ABMD      Abiomed  Health Care   Danvers, Massachusetts  31.05.2018   

       Founded  
0         1902  
1         1916  
2         1888  
3  2013 (1888)  
4         1981

In [97]:
sap.describe()

Symbol Security             GICS_Sector                       hq  \
count     458      458                     458                      458   
unique    458      458                      11                      243   
top       MMM       3M  Information Technology  New York City, New York   
freq        1        1                      68                       34   

          included Founded  
count          458     458  
unique         351     186  
top     04.03.1957    1998  
freq            37      10

In [99]:
sap['Founded'].describe()

count      458
unique     186
top       1998
freq        10
Name: Founded, dtype: object

In [102]:
oldest = min(sap.included)
oldest


'01.01.1994'

In [ ]:
youngest = max(sap.included)
youngest 

2.2. we successfully deleted those firms/rows, here i tried to find the youngest and the oldest